tmt/options.py

```diff
@@ -66,6 +109,22 @@ def _check_method(self, args):
                 elif args[index].startswith('-h'):
                     how = re.sub('^-h ?', '', args[index])
                     break
+                # Do not check how of other methods if chaining is used
+                # (e.g. tmt run discover provision -h local)
+                elif not upcoming_value and args[index] in tmt.steps.STEPS:
```

Comment:
I am not sure if the proposed way to check for possible upcoming value works. Tested with:

    tmt run prepare -p discover -h install

The `self.params` contain in this case:

    [<Option how>, <Option verbose>, <Option debug>, <Option quiet>, <Option force>, <Option dry>]

These are just generic step flags and none of them would match the condition below. The real `prepare` options cannot be known at this time as we have not yet decided the step method.

I'm afraid we can only blindly check for the arg name and stop at the first suspicious value. Which should include other run subcommands `login`, `reboot`, `plans` and `tests` as well.

Reply:
Proposing to simplify the option search by c3c549e. Please, have a look and let me know if I overlooked anything important.

Reply:
Yes, you are right, it won't work in this case because `-p` is not a valid option of `prepare` so the for loop does nothing (no break) -> jumps into the else and a value is not expected in the next iteration, which obviously is incorrect...

Your changes make sense as a simplification, though this won't be perfect (and I don't think we can achieve perfection in this case, since we are hacking around click's parsing :/ ) - for example the checking will stop if a subcommand is used as a value of a flag, imagine something like `tmt run provision -h a --something test`. In such case, test is a subcommand and the search for how will stop even if it should not. But I guess that's fine since it was specified beforehand, so there may not actually be a problem...

I agree with the changes, lets merge if the tests pass (not once was I surprised during this pull request with a random test failing out of nowhere :) )

Reply:
Yeah, not ideal, but good enough for now, I'd say. Let's keep something for the future ;-) Thanks for reviewing the changes.